src/main/msc/emfat.c

```diff
@@ -0,0 +1,764 @@
+    }
+
+    le = emfat->priv.last_entry;
+    while (count != 0) {
+        if (!IS_CLUST_OF(curr, le)) {
+            le = find_entry(emfat, curr, le);
+            if (le == NULL) {
+                le = emfat->priv.last_entry;
+                *values = CLUST_RESERVED;
+                values++;
+                count--;
+                curr++;
+                continue;
+            }
+        }
+        if (le->dir) {
+            if (curr == le->priv.last_clust) {
+                *values = CLUST_EOF;
+            } else {
+                *values = curr + 1;
+            }
+        } else {
+            if (curr == le->priv.last_clust) {
+                *values = CLUST_EOF;
+            } else if (curr > le->priv.last_clust) {
+                *values = CLUST_FREE;
+            } else {
+                *values = curr + 1;
+            }
+        }
+        values++;
+        count--;
+        curr++;
+    }
+    emfat->priv.last_entry = le;
+}
+
+void fill_entry(dir_entry *entry, const char *name, uint8_t attr, uint32_t clust, const uint32_t cma[3], uint32_t size)
+{
+    int i, l, l1, l2;
+    int dot_pos;
+
+    memset(entry, 0, sizeof(dir_entry));
+
+    if (cma) {
+        entry->crt_date = cma[0] >> 16;
+        entry->crt_time = cma[0] & 0xFFFF;
+        entry->lst_mod_date = cma[1] >> 16;
+        entry->lst_mod_time = cma[1] & 0xFFFF;
+        entry->lst_access_date = cma[2] >> 16;
+    }
+
+    l = strlen(name);
+    dot_pos = -1;
+    if ((attr & ATTR_DIR) == 0) {
+        for (i = l - 1; i >= 0; i--) {
+            if (name[i] == '.')
+            {
+                dot_pos = i;
+                break;
+            }
+        }
+    }
+    if (dot_pos == -1) {
+        l1 = l > FILE_NAME_SHRT_LEN ? FILE_NAME_SHRT_LEN : l;
+        l2 = 0;
+    } else {
+        l1 = dot_pos;
+        l1 = l1 > FILE_NAME_SHRT_LEN ? FILE_NAME_SHRT_LEN : l1;
+        l2 = l - dot_pos - 1;
+        l2 = l2 > FILE_NAME_EXTN_LEN ? FILE_NAME_EXTN_LEN : l2;
+    }
+    memset(entry->name, ' ', FILE_NAME_SHRT_LEN + FILE_NAME_EXTN_LEN);
+    memcpy(entry->name, name, l1);
+    memcpy(entry->extn, name + dot_pos + 1, l2);
+    for (i = 0; i < FILE_NAME_SHRT_LEN; i++) {
+        if (entry->name[i] >= 'a' && entry->name[i] <= 'z') {
+            entry->name[i] -= 0x20;
+        }
+    }
+    for (i = 0; i < FILE_NAME_EXTN_LEN; i++) {
+        if (entry->extn[i] >= 'a' && entry->extn[i] <= 'z') {
+            entry->extn[i] -= 0x20;
+        }
+    }
+
+    entry->attr = attr;
+    entry->reserved = 24;
+    entry->strt_clus_hword = clust >> 16;
+    entry->strt_clus_lword = clust;
+    entry->size = size;
+    return;
+}
+
+void fill_dir_sector(emfat_t *emfat, uint8_t *data, emfat_entry_t *entry, uint32_t rel_sect)
+{
+    dir_entry *de;
+    uint32_t avail;
+
+    memset(data, 0, SECT);
+    de = (dir_entry *)data;
+    avail = SECT;
+
+    if (rel_sect == 0) { // 1. first sector of directory
+        if (entry->priv.top == NULL)
+        {
```

Comment:
Braces on newlines.

Reply:
It's another formatting/style problem. I think I will do full tidying on emfat files for the following reasons.
- The original project can be considered finished and the repo is stale.
- There were extremely non-standard style such as
```
        if (curr == le->priv.last_clust)
                *values = CLUST_EOF; else
                *values = curr + 1;
```